proknow/CustomMetrics.py

```diff
@@ -181,7 +181,14 @@ def resolve_by_name(self, name):
         """
         assert isinstance(name, six.string_types), "`name` is required as a string."
 
-        custom_metric = self.find(name=name)
+        if self._cache is None:
+            self.query()
+        for metric in self._cache:
+            if metric.name.lower() == name.lower():
```

Comment:
Not that it is a huge performance improvement, but you might want to move `name.lower()` outside of the loop and declare as a local variable so it doesn't have to be computed each time.

Reply:
Fixed